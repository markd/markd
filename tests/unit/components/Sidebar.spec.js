import { shallowMount, createLocalVue } from '@vue/test-utils'
import Component from '@/components/Sidebar.vue'

import Vuex from 'vuex'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(VueRouter)

const title = 'markd'

const store = new Vuex.Store({
  state: {
    title,
    pinned: [
      {
        icon: 'cloud_queue',
        name: 'All Bookmarks'
      },
      {
        icon: 'favorite_border',
        name: 'Favourite Bookmarks'
      }
    ],
    collections: [
      {
        colour: 'red',
        name: 'Web Development'
      },
      {
        colour: 'green',
        name: 'Spicy Memes'
      }
    ]
  }
})

describe('Sidebar.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(Component, {
      localVue,
      store
    })

    expect(wrapper).toMatchSnapshot()
  })

  it('display title', () => {
    const wrapper = shallowMount(Component, {
      localVue,
      store
    })

    expect(wrapper.find('.sidebar__title').text()).toBe(title)
  })
})
