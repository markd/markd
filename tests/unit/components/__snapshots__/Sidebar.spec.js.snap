// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar.vue renders 1`] = `
<div class="sidebar">
  <div class="sidebar__header">
    <h1 class="sidebar__title">markd</h1>
    <router-link-stub to="/add" tag="a" event="click" class="sidebar__addBookmark"><svg class="sidebar__addBookmarkIcon">
        <use xlink:href="#icon-plus"></use>
      </svg></router-link-stub>
  </div>
  <div class="sidebar__content">
    <sidebargroup-stub title="Pinned">
      <pinned-stub></pinned-stub>
    </sidebargroup-stub>
    <sidebargroup-stub title="Collections">
      <collections-stub></collections-stub>
    </sidebargroup-stub>
    <sidebargroup-stub title="Tags">
      <tags-stub></tags-stub>
    </sidebargroup-stub>
  </div>
</div>
`;
